tests/unit/test_repos_repo.py

```diff
@@ -61,6 +66,48 @@ def test_asset_requires_a_positive_id(self):
 
         assert self.session.get.called is False
 
+    def test_create_file(self):
+        """Verify the request for creating a file on a repository."""
+        data = {
+            'path': 'hello.txt',
+            'message': 'my commit message',
+            'content': b'bXkgbmV3IGZpbGUgY29udGVudHM=',
+            'committer': {
+                'name': 'Scott Chacon',
+                'email': 'schacon@gmail.com'
+            }
+        }
+
+        with mock.patch.object(GitHubCore, '_json') as _json:
```

Comment:
@sigmavirus24 I know we want to limit using mock however `create_file` inspects the response (snippet below):

```
        json = None                                                                 
        if path and message and content:                                            
            url = self._build_url('contents', path, base_url=self._api)             
            content = b64encode(content).decode('utf-8')                            
            data = {'message': message, 'content': content, 'branch': branch,   
                    'committer': validate_commmitter(committer),                    
                    'author': validate_commmitter(author)}                          
            self._remove_none(data)                                                 
            json = self._json(self._put(url, data=dumps(data)), 201)                
            if 'content' in json and 'commit' in json:                              
                json['content'] = Contents(json['content'], self)                   
                json['commit'] = Commit(json['commit'], self)                       
        return json  
```

Let me know if you have any suggestions. Also, should we be mocking `_remove_none` or just rely on asserting on the session?